datafusion/core/src/physical_optimizer/aggregate_statistics.rs

```diff
@@ -148,10 +152,10 @@ fn take_optimizable_table_count(
                 .as_any()
                 .downcast_ref::<expressions::Literal>()
             {
-                if lit_expr.value() == &ScalarValue::UInt8(Some(1)) {
+                if lit_expr.value() == &COUNT_STAR_EXPANSION {
                     return Some((
-                        ScalarValue::UInt64(Some(num_rows as u64)),
-                        "COUNT(UInt8(1))",
+                        ScalarValue::Int64(Some(num_rows as i64)),
```

Comment:
The change from `UInt64` to `Int64` here and a few lines below is the actual bug fix / change of behavior -- the rest of this PR is testing / readability improvements